components/hab/src/command/pkg/bulkupload.rs

```diff
@@ -0,0 +1,126 @@
+//! Uploads packages from cache to a [Depot](../depot).
+//!
+//! # Examples
+//!
+//! ```bash
+//! $ hab pkg bulkupload /path/to/artifact_download_dir \
+//!     -u http://localhost:9632
+//! ```
+//!
+//! Will upload all packages in cache to Builder.
+
+use crate::{api_client::{self,
+                         BuildOnUpload,
+                         Client},
+            command,
+            common::ui::{Glyph,
+                         Status,
+                         UIReader,
+                         UIWriter,
+                         UI},
+            error::{Error,
+                    Result},
+            hcore::{package::PackageArchive,
+                    ChannelIdent},
+            PRODUCT,
+            VERSION};
+use glob::glob_with;
+use reqwest::StatusCode;
+use std::{collections::BTreeSet,
+          path::{Path,
+                 PathBuf}};
+
+/// Bulk Upload the packages from the cache to a Depot.
+///
+/// # Failures
+///
+/// * Fails if it cannot create a missing origin
+/// * Fails if it cannot upload the artifact
+#[allow(clippy::too_many_arguments)]
+pub fn start(ui: &mut UI,
+             bldr_url: &str,
+             additional_release_channel: &Option<ChannelIdent>,
+             token: &str,
+             artifact_path: &Path,
+             force_upload: bool,
+             auto_build: BuildOnUpload,
+             auto_create_origins: bool,
+             key_path: &Path)
+             -> Result<()> {
+    const OPTIONS: glob::MatchOptions = glob::MatchOptions { case_sensitive:              true,
+                                                             require_literal_separator:   true,
+                                                             require_literal_leading_dot: true, };
+    let artifact_paths =
+        vec_from_glob_with(&artifact_path.join("*.hart").display().to_string(), OPTIONS);
+
+    ui.begin(format!("Preparing to upload artifacts to the '{}' channel on {}",
+                     additional_release_channel.clone()
+                                               .unwrap_or_else(ChannelIdent::unstable),
+                     bldr_url))?;
+    ui.status(Status::Using,
+              format!("{} for artifacts and {} for signing keys.",
+                      &artifact_path.display(),
+                      key_path.display()))?;
+    ui.status(Status::Found,
+              format!("{} artifact(s) for upload.", artifact_paths.len()))?;
+    ui.status(Status::Discovering,
+              String::from("origin names from local artifact cache.."))?;
+
+    let mut origins = BTreeSet::new();
+    for artifact_path in &artifact_paths {
+        let ident = PackageArchive::new(&artifact_path).ident()?;
+        origins.insert(ident.origin);
+    }
+    let mut origins_to_create: Vec<String> = Vec::new();
+    let api_client = Client::new(bldr_url, PRODUCT, VERSION, None)?;
+    for origin in origins {
```

Comment:
Super tiny nit - add a whitespace blank line to break up the code sections a bit more (readability)

Reply:
Thank you: addressed in https://github.com/habitat-sh/habitat/pull/7133/commits/d13255c9d570503e6694cb0975e82350426e5314